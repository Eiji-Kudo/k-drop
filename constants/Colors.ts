const tintColorLight = '#0a7ea4'

export const Colors = {
  text: '#11181C',
  background: '#FAF0F4',
  tint: tintColorLight,
  icon: '#687076',
  tabIconDefault: '#687076',
  tabIconSelected: tintColorLight,
  border: '#F0C2D9',
  primary: '#FF9FCC',
  secondary: '#FFD9EB',
  tertiary: '#B52E6E',
}
