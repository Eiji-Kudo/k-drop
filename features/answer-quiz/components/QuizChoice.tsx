import { ThemedText } from '@/components/ThemedText'
import { Colors } from '@/constants/Colors'
import { QuizVariant } from '@/features/answer-quiz/constants/quizVariant'
import { Pressable, PressableProps, StyleSheet, ViewStyle } from 'react-native'

type Props = PressableProps & {
  index: number
  label: string
  variant: QuizVariant
  style?: ViewStyle
}

export function QuizChoice({ index, label, variant, style, ...rest }: Props) {
  return (
    <Pressable
      {...rest}
      style={[
        styles.choiceButton,
        variant === QuizVariant.CORRECT && styles.choiceButtonCorrect,
        variant === QuizVariant.INCORRECT && styles.choiceButtonIncorrect,
        style,
      ]}
    >
      <ThemedText
        style={styles.choiceText}
      >{`${index + 1}. ${label}`}</ThemedText>
    </Pressable>
  )
}

const styles = StyleSheet.create({
  choiceButton: {
    borderColor: Colors.border,
    borderRadius: 8,
    borderWidth: 1,
    paddingHorizontal: 16,
    paddingVertical: 12,
  },
  choiceButtonCorrect: {
    backgroundColor: Colors.secondary,
    borderColor: Colors.primary,
  },
  choiceButtonIncorrect: {
    backgroundColor: Colors.toastError,
    borderColor: Colors.toastError,
  },
  choiceText: { fontSize: 16, fontWeight: '500' },
})
