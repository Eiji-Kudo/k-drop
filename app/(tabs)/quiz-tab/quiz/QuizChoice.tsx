import { ThemedText } from '@/components/ThemedText'
import { Pressable, PressableProps, StyleSheet, ViewStyle } from 'react-native'

type Variant = 'default' | 'correct' | 'wrong'

type Props = PressableProps & {
  index: number
  label: string
  variant: Variant
  style?: ViewStyle
}

export function QuizChoice({ index, label, variant, style, ...rest }: Props) {
  return (
    <Pressable
      {...rest}
      style={[
        styles.choiceButton,
        variant === 'correct' && styles.choiceButtonCorrect,
        variant === 'wrong' && styles.choiceButtonWrong,
        style,
      ]}
    >
      <ThemedText style={styles.choiceText}>{`${index + 1}. ${label}`}</ThemedText>
    </Pressable>
  )
}

const styles = StyleSheet.create({
  choiceButton: {
    borderColor: '#d1d1d1',
    borderRadius: 8,
    borderWidth: 1,
    paddingHorizontal: 16,
    paddingVertical: 12,
  },
  choiceButtonCorrect: {
    backgroundColor: '#e4f5ff',
    borderColor: '#68c1f1',
  },
  choiceButtonWrong: {
    backgroundColor: '#ffecec',
    borderColor: '#f56f6f',
  },
  choiceText: { fontSize: 16, fontWeight: '500' },
})
