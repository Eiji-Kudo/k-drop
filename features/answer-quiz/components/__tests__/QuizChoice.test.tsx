import React from 'react'
import { render, fireEvent } from '@testing-library/react-native'
import { QuizChoice } from '../QuizChoice'
import { QuizVariant } from '../../constants/quizVariant'

describe('QuizChoice', () => {
  const onPressMock = jest.fn()

  beforeEach(() => {
    jest.clearAllMocks()
  })

  it('renders correctly with the right label', () => {
    const { getByText, getByTestId } = render(
      <QuizChoice
        index={0}
        label="Test Option"
        variant={QuizVariant.UNANSWERED}
        onPress={onPressMock}
      />
    )

    // Check if button is rendered with correct text (index + 1 prefixed)
    expect(getByText('1. Test Option')).toBeTruthy()
    expect(getByTestId('quiz-choice')).toBeTruthy()
  })

  it('calls onPress when pressed', () => {
    const { getByTestId } = render(
      <QuizChoice
        index={0}
        label="Test Option"
        variant={QuizVariant.UNANSWERED}
        onPress={onPressMock}
      />
    )

    // Press the choice
    fireEvent.press(getByTestId('quiz-choice'))

    // Check if onPress was called
    expect(onPressMock).toHaveBeenCalledTimes(1)
  })

  it('applies correct styling when answer is correct', () => {
    const { getByTestId } = render(
      <QuizChoice
        index={0}
        label="Test Option"
        variant={QuizVariant.CORRECT}
        onPress={onPressMock}
      />
    )

    // In React Native Testing Library, we can't easily check styles directly
    // We would need to use snapshot testing for this, but for now we'll just
    // verify that the component renders with the correct variant
    const choiceButton = getByTestId('quiz-choice')
    expect(choiceButton).toBeTruthy()
  })

  it('applies incorrect styling when answer is incorrect', () => {
    const { getByTestId } = render(
      <QuizChoice
        index={0}
        label="Test Option"
        variant={QuizVariant.INCORRECT}
        onPress={onPressMock}
      />
    )

    const choiceButton = getByTestId('quiz-choice')
    expect(choiceButton).toBeTruthy()
  })

  it('does not call onPress when disabled', () => {
    const { getByTestId } = render(
      <QuizChoice
        index={0}
        label="Test Option"
        variant={QuizVariant.UNANSWERED}
        disabled={true}
        onPress={onPressMock}
      />
    )

    // Press the choice
    fireEvent.press(getByTestId('quiz-choice'))

    // Check that onPress was not called due to disabled state
    expect(onPressMock).not.toHaveBeenCalled()
  })
})