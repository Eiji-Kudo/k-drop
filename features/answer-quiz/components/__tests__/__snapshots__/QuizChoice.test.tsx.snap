// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuizChoice applies correct styling when answer is correct 1`] = `
<div
  data-testid="quiz-choice"
  onClick={[MockFunction]}
  style={
    [
      {
        "borderColor": "#CCCCCC",
        "borderRadius": 8,
        "borderWidth": 1,
        "paddingHorizontal": 16,
        "paddingVertical": 12,
      },
      {
        "backgroundColor": "#4CD964",
        "borderColor": "#007AFF",
      },
      false,
      undefined,
    ]
  }
  testID="quiz-choice"
>
  1. Test Option
</div>
`;

exports[`QuizChoice applies incorrect styling when answer is incorrect 1`] = `
<div
  data-testid="quiz-choice"
  onClick={[MockFunction]}
  style={
    [
      {
        "borderColor": "#CCCCCC",
        "borderRadius": 8,
        "borderWidth": 1,
        "paddingHorizontal": 16,
        "paddingVertical": 12,
      },
      false,
      {
        "backgroundColor": "#FF3B30",
        "borderColor": "#FF3B30",
      },
      undefined,
    ]
  }
  testID="quiz-choice"
>
  1. Test Option
</div>
`;

exports[`QuizChoice renders correctly with the right label 1`] = `
<div
  data-testid="quiz-choice"
  onClick={[MockFunction]}
  style={
    [
      {
        "borderColor": "#CCCCCC",
        "borderRadius": 8,
        "borderWidth": 1,
        "paddingHorizontal": 16,
        "paddingVertical": 12,
      },
      false,
      false,
      undefined,
    ]
  }
  testID="quiz-choice"
>
  1. Test Option
</div>
`;
